src/core/config/parent_search.h

```diff
@@ -40,16 +40,29 @@
  *
  * Define as 1 to enable periodic parent search feature.
  *
- * When this feature is enabled an end-device/child (while staying attached) will periodically search for a possible
- * better parent and will switch parent if a better one is found.
+ * When this feature is enabled, an end device/child (while staying attached) periodically searches for a possible
+ * better parent and switches parents if a better one is found.
  *
- * The child will periodically check the average RSS value for the current parent, and only if it is below a specific
- * threshold, a parent search is performed. The `OPENTHREAD_CONFIG_PARENT_SEARCH_CHECK_INTERVAL` specifies the
- * check interval (in seconds) and `OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD` gives the RSS threshold.
+ * Every `OPENTHREAD_CONFIG_PARENT_SEARCH_CHECK_INTERVAL` seconds, the child checks its average RSS to its current
+ * parent. Only if the average RSS is below a configured threshold (`OPENTHREAD_CONFIG_PARENT_SEARCH_RSS_THRESHOLD`)
+ * does the device start a parent search process.
  *
- * Since the parent search process can be power consuming (child needs to stays in RX mode to collect parent response)
- * and to limit its impact on battery-powered devices, after a parent search is triggered, the child will not trigger
- * another one before a specified backoff interval specified by `OPENTHREAD_CONFIG_PARENT_SEARCH_BACKOFF_INTERVAL`.
+ * The parent search mechanism depends on whether the device is an FTD child or an MTD child.
+ *
+ * An FTD device receives and processes MLE Advertisements from neighboring routers. It uses this information to track
+ * which neighboring routers can be selected as potential new parents and to track the link quality to each.
```

Comment:
Suggested clarification that it tracks one-way link quality. This is useful since in Thread's terminology, the Link Quality (LQ) categories are about two-way link quality which is not the thing being tracked if I'm correct.

```suggestion
 * which neighboring routers can be selected as potential new parents and to track the one-way link quality to each.
```